modules/grandpa/src/lib.rs

```diff
@@ -335,20 +362,19 @@ pub mod pallet {
 		}
 	}
 
-	/// Number mandatory headers that we may accept in the current block for free (returning
-	/// `Pays::No`).
+	/// Number of free header submissions that we may yet accept in the current block.
 	///
-	/// If the `FreeMandatoryHeadersRemaining` hits zero, all following mandatory headers in the
+	/// If the `FreeHeadersRemaining` hits zero, all following mandatory headers in the
 	/// current block are accepted with fee (`Pays::Yes` is returned).
 	///
-	/// The `FreeMandatoryHeadersRemaining` is an ephemeral value that is set to
-	/// `MaxFreeMandatoryHeadersPerBlock` at each block initialization and is killed on block
+	/// The `FreeHeadersRemaining` is an ephemeral value that is set to
+	/// `MaxFreeHeadersPerBlock` at each block initialization and is killed on block
 	/// finalization. So it never ends up in the storage trie.
 	#[pallet::storage]
 	#[pallet::whitelist_storage]
 	#[pallet::getter(fn free_mandatory_headers_remaining)]
-	pub(super) type FreeMandatoryHeadersRemaining<T: Config<I>, I: 'static = ()> =
-		StorageValue<_, u32, ValueQuery>;
+	pub type FreeHeadersRemaining<T: Config<I>, I: 'static = ()> =
```

Comment:
I think this could change storage keys, right?

Regardless of this issue, shouldn't we setup `storage_version` for all our pallets to be prepared for migrations?
Anyway, we will need it for https://github.com/paritytech/parity-bridges-common/issues/1869.
I could do it as a separate issue/PR - prepare bridge pallets for migrations. @acatangiu @svyatonik  Wdyt?

Should be something like:
```
        #[pallet::pallet]
	#[pallet::storage_version(STORAGE_VERSION)]
	pub struct Pallet<T, I = ()>(PhantomData<(T, I)>);
```

Actually, we are on the version `0`

![image](https://github.com/paritytech/parity-bridges-common/assets/8159517/4162c852-29ca-4845-bea2-beb1208061c7)

Reply:
ok, will do here: https://github.com/paritytech/parity-bridges-common/issues/2947

Reply:
> It is there just to make migrations safer, right (make them idempotent)? 

Yes, afaik. I played a little bit with migrations when fixing fellows bump.
Anyway, there is a CI pipeline "check-migrations", and I think it should detect undecodable storage keys, so if we change stored keys for existing values, the CI should detect that and fail, so we know that we have to add migration and so on.